### PR TITLE
RequirementCategory: add 'requirement_add' command

### LINKED ISSUE #73: As a user, I want to be able to add modules into requirement categories
... so that I am able to classify the modules into the corresponding requirement category

### PR DESCRIPTION
**Pull request by chyeo**
Resolves #73.

The application currently does not allow users to add an existing module to a requirement category. 

Let's add a command to allow users to be able to add a module to a requirement category.

Let's also add testcases to see the response of the application if undesired inputs were made.

The following are commits made to implement the new command:
* [1/8] [SampleDataUtil: update data in preparation for new command](https://github.com/CS2113-AY1819S2-T09-1/main/pull/82/commits/ca9f1a66ad551be9dfe3383c5a513f42f7cc7c1c)
* [2/8] [RequirementCategoryBuilder: update to faciliate new command](https://github.com/CS2113-AY1819S2-T09-1/main/pull/82/commits/fc44d7f10ad53e5430c0c43173979450beb3c6d0)
* [3/8] [commands: add 'RequirementAddCommand' class with uninitialized methods](https://github.com/CS2113-AY1819S2-T09-1/main/pull/82/commits/b583898cc376de0878c01a96dad70972b2646594)
* [4/8] [commands: initialize methods created in 'RequirementAddCommand' class](https://github.com/CS2113-AY1819S2-T09-1/main/pull/82/commits/b6eae3fa3d0ec93da1ed23948a8c5115db09eef6)
* [5/8] [UniqueRequirementCategoryList: implement logic of initialized methods](https://github.com/CS2113-AY1819S2-T09-1/main/pull/82/commits/0bc2c77359fbd657578889e635dafe7103bdfbf4)
* [6/8] [parser: implementing 'RequirementAddCommandParser' for the new command](https://github.com/CS2113-AY1819S2-T09-1/main/pull/82/commits/0f1401f3a976345ab2e1469fd083c9372a592d30)
* [7/8] [AddressBookParser: update to allow application to take in new command](https://github.com/CS2113-AY1819S2-T09-1/main/pull/82/commits/bfe8f3d5a6ce2abdd52d9efefb5e3ff2be3dba5e)
* [8/8] [test: create new testcases for the implemented command](https://github.com/CS2113-AY1819S2-T09-1/main/pull/82/commits/bfd61c624df5f84f22feb3a1c7aaff33ec14af7c)

### CODE REVIEW

File: src/main/java/seedu/address/model/requirement/UniqueRequirementCategoryList.java [hunk 1]
```diff
@@ -54,6 +74,69 @@ public void add(RequirementCategory toAdd) {
...
+                new RequirementCategory(internalList.get(location).getName(), internalList.get(location).getCredits(),
+                        inputList);
+
+        setRequirementCategory(internalList.get(location), edited);
+    }
+
+    /**
+     * Returns true if the list contains an equivalent requirement as the given argument.
+     */
+    public boolean hasModuleInRequirementCategory(RequirementCategory toCheck) {
```
Comment (lycjackie) [inline src/main/java/seedu/address/model/requirement/UniqueRequirementCategoryList.java]: `hasModuleInRequirementCategory ` but no module is used? 

Consider changing the method name

File: src/test/java/seedu/address/logic/commands/RequirementAddCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                        .toModelType(), new DegreePlannerList(), new UserPrefs());
+    }
+
+    @Test public void constructor_nullModule_throwsNullPointerException() {
+        thrown.expect(NullPointerException.class);
+        new RequirementAddCommand(null);
+    }
+
+    @Test public void execute_requirementCategoryDoesNotExist_throwsCommandException() {
+        Set<Code> codeList = new HashSet<>();
```
Comment (lycjackie) [inline src/test/java/seedu/address/logic/commands/RequirementAddCommandTest.java]: Do you have to initialize this `codeList` always? 

Consider putting it as a global variable, clear it when needed.

File: src/main/java/seedu/address/model/requirement/UniqueRequirementCategoryList.java [hunk 3]
```diff
@@ -54,6 +74,69 @@ public void add(RequirementCategory toAdd) {
         internalList.add(toAdd);
     }
 
+    /**
+     * Adds a module to a requirement category.
+     * The module must not already exist in the requirement category.
+     */
+    public void addModuleToRequirementCategory(RequirementCategory toAdd, Model model) {
```
Comment (lycjackie) [inline src/main/java/seedu/address/model/requirement/UniqueRequirementCategoryList.java]: `model` is not used anywhere in this method. 

Why are you taking in this extra parameter?

**Review by lycjackie — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/AddressBook.java [hunk 4]
```diff
@@ -154,6 +154,32 @@ public void removeRequirementCategory(RequirementCategory key) {
...
+    public boolean hasModuleInRequirementCategory(RequirementCategory requirementCategory) {
+        requireNonNull(requirementCategory);
+        return requirementCategories.hasModuleInRequirementCategory(requirementCategory);
+    }
+
+    /**
+     * Returns false if a module with the same identity as {@code requirementCategory} does not exists
+     * in the current moduleList.
+     */
+    public boolean doesModuleExist(RequirementCategory requirementCategory, Model model) {
```
Comment (lycjackie) [inline src/main/java/seedu/address/model/AddressBook.java]: This method seems to be similar to 
https://github.com/CS2113-AY1819S2-T09-1/main/blob/b6eae3fa3d0ec93da1ed23948a8c5115db09eef6/src/main/java/seedu/address/model/AddressBook.java#L166-L173

Based on the comments. 

I understand that this method, you are checking the `Code` params instead. 
Consider the changing the method name to make it clearer and update the
comments.

For now I will accept that you are taking in `Model model` due to the lack
of util method in checking if a module `hasModuleCode`

However, this should be updated once #91 is merge in.

**Review by lycjackie — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/requirement/UniqueRequirementCategoryList.java (hunk 1 above)
Comment (chyeo) [reply src/main/java/seedu/address/model/requirement/UniqueRequirementCategoryList.java]: Roger, have removed the unused parameter and changed the method name

File: src/main/java/seedu/address/model/requirement/UniqueRequirementCategoryList.java (hunk 3 above)
Comment (chyeo) [reply src/main/java/seedu/address/model/requirement/UniqueRequirementCategoryList.java]: Roger, have removed the unused parameter

File: src/test/java/seedu/address/logic/commands/RequirementAddCommandTest.java (hunk 2 above)
Comment (chyeo) [reply src/test/java/seedu/address/logic/commands/RequirementAddCommandTest.java]: Have made it to a global variable and cleared it before use

File: src/main/java/seedu/address/model/AddressBook.java (hunk 4 above)
Comment (chyeo) [reply src/main/java/seedu/address/model/AddressBook.java]: Roger, have added a todo comment near the method, will update it again once #91 have been merged in

**Review by lycjackie — APPROVED** [review]

**Comment by lycjackie** [comment]
Note: 

Although there were checks that failed, `travis` still managed to pass,
hence this PR was merged in.

Appveyor checks were throttled. Thus we ignored it.

**Pull request merged**